src/test/java/utilities/HttpJsonServer.java

```diff
@@ -0,0 +1,41 @@
+package utilities;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class HttpJsonServer {
```

Comment:
I would really just mock the objects to return the json directly. An http server is overkill.

Reply:
I agree that add some complexity but how would you mock it using a groovy script?
When mocking the objects, you'll need to instantiate them and pass them to the groovy script. 

IMHO, the tests are covering a large part of the implementation with minimum effort (since we took it from the existing extension)

Reply:
There is a test already that mocks VaultAccessor for a pipeline script, in the repo

Reply:
After discussion, we agreed that having the web server help to cover broader use cases and we want to limit the time invested into coding for the plugin.